File: src/main/java/ru/anarcom/octopus/model/Role.java
package ru.anarcom.octopus.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * Simple domain object that represents application user's role - ADMIN, USER, etc.
 *
 * @author Eugene Suleimanov
 * @version 1.0
 */

@EqualsAndHashCode(callSuper = true)
@Data
public class Role extends BaseEntity {

    private String name;
//    private List<User> users;

}

File: src/main/kotlin/ru/anarcom/octopus/entity/Role.kt
package ru.anarcom.octopus.entity

import lombok.Data
import lombok.EqualsAndHashCode

/**
 * Simple Role object. In future, it will be base for ABAC access model.
 */
@EqualsAndHashCode(callSuper = true)
@Data
class Role : BaseEntity() {
    var name: String = ""
}